### ISSUE TITLE
Blur bleeding

### ISSUE THREAD

**Issue by Laks0** [ibhagwan/picom#12]
### Platform
Arch Linux 5.4.65 LTS 

### GPU, drivers, and screen setup

Two monitors side by side set up with lastest Nouveau and Intel free drivers. Nvidia Geforce 1050

### Environment

I3-gaps with polybar

### Problem
I don't know if this is a general Picom issue or if it is related to this fork but I get a lot of bleeding from blured windows, There's a white line to the side my other monitor is and a black line to the top.
Here's a picture of how it look, and also my config file.
![img-2020-09-16-194302](https://user-images.githubusercontent.com/35859490/93400614-8f1ade00-f856-11ea-84fc-64b2e2a15beb.png)
Without blur there's no difference between the color of polybar and the terminal. The black line goes away if I kill polybar.

**Comment by Azorlogh** [comment]
I have the same problem, my windows bleed into each other.
See:
![picom_blur_bleeding](https://user-images.githubusercontent.com/17968319/95017718-4edf9c00-065b-11eb-91c0-c1485715484b.png)
The left side is darkened by a neighboring window.

**Comment by robbins** [comment]
Same here, but only with experimental-backends enabled. I have an xfce panel with transparency and blur, and it blurs only my wallpaper behind it with nothing else open, but with any program, eg Microsoft Edge near the bar, there's a white bleed into the blur.

**Comment by ibhagwan** [comment]
The code is currently being merged into the main picom branch by yshui, you can follow the work [in this pull request](https://github.com/yshui/picom/pull/558), once the fork is complete we can revisit this issue.

There's a good chance this will be solved as there have been many changes to the main development branch since I created this fork.

**Comment by tryone144** [comment]
This "bleeding" is an intentional design choice.
The algorithm actually blurs a larger area than the window to prevent the background rapidly changing its color at the edge. See https://github.com/yshui/picom/pull/382#issuecomment-617930559 for a more detailed explanation.
A better/"fixed" implementation of `--transparent-clipping` could possibly be used to restrict some windows (like the bar) to only blur the wallpaper. This behaviour would probably be preferable for tiling window managers. As of now, this is not possible.

**Comment by ibhagwan** [comment]
Tysm for the explanation @tryone144!

**Comment by Neightro** [comment]
I've experienced this issue as well at the edges of screens on a mutlihead setup. The monitors are different sizes, so they leave some blank space within Xorg's coordinate system, which apparently defaults to white.

As a workaround, I've noticed that by using xrandr to briefly cause one of the outputs to render into this space, that the results can be made much more pleasing. The ideal solution would probably be clipping for individual monitor outputs, though I'd imagine that's impossible right now for the same reasons as window clipping.

**Comment by ibhagwan** [comment]
I guess this issue could be closed for the time being, the code was rebased and up to date with mainline picom, since the kawase blur on experimental is now part of the main fork if there are any changes that affect this specific issue they’ll naturally resolve themselves with the next rebase.
